packages/connection/index.js

```diff
@@ -106,49 +107,50 @@ class Connection extends EventEmitter {
     return this.jid
   }
 
-  _online() {
-    this.emit('online', this.jid)
-  }
-
-  _authenticated() {
-    this.emit('authenticated')
+  _status(status, ...args) {
+    this.status = status
+    this.emit('status', status, ...args)
+    this.emit(status, ...args)
```

Comment:
Not sure I am getting my point across but here:

For things that are "state" I would always want events. A connection can be "online" or "offline" or "connecting" or whatever. For those I would use 
``client.on('online', () => ...``

For things that are actions that can succeed or fail I would always use promises. Hence I wouldn't do 
``client.when('authenticated')`` or ``client.when('online')`` but rather
``client.authenticate(creds).then()``

I am not going to argue further on this point as I have a limited amount of time I can dedicate to this and therefore it's unfair I get my say on it. No offence meant or taken :)